<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1524</width>
    <height>712</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QFrame" name="entriesFrame">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QFormLayout" name="entriesControlLayout">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="entriesLabel">
               <property name="text">
                <string>Entries </string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="entriesComboBox">
               <property name="maxVisibleItems">
                <number>20</number>
               </property>
               <item>
                <property name="text">
                 <string>All</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="EntriesListView" name="entriestListView">
             <property name="styleSheet">
              <string notr="true">QListWidget {
alternate-background-color: #EDEDED;
background-color: white;
}</string>
             </property>
             <property name="horizontalScrollBarPolicy">
              <enum>Qt::ScrollBarAlwaysOff</enum>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QFrame" name="attributesFrame">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="attributesControlsLayout">
             <item>
              <widget class="QLabel" name="attributesLabel">
               <property name="text">
                <string>Found attributes of entry </string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QToolButton" name="toolButton">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Run extraction for current entry and update attributes. Handy for testing if rawtext editing has helped.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>Rerun analysis</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="EntryTableView" name="tableView">
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
             <attribute name="horizontalHeaderCascadingSectionResizes">
              <bool>false</bool>
             </attribute>
             <attribute name="horizontalHeaderStretchLastSection">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QFrame" name="currentFrame">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="spacing">
              <number>2</number>
             </property>
             <item>
              <widget class="QLabel" name="rawTextLabel">
               <property name="text">
                <string>Rawtext </string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="rawTextCurrentLabel">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="rawTexthorizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QToolButton" name="combineButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="autoFillBackground">
                <bool>false</bool>
               </property>
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>Combine</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="rawTextSaveButton">
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>Save</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="rawTextDeleteButton">
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>Delete</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="rawTextTextEdit"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QFrame" name="previousFrame">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <property name="bottomMargin">
            <number>12</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="previousControlsLayout">
             <property name="spacing">
              <number>2</number>
             </property>
             <property name="sizeConstraint">
              <enum>QLayout::SetDefaultConstraint</enum>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="previousEntryLabel">
               <property name="text">
                <string>Previous entry</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="previousHorizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QToolButton" name="previousENtrySaveButton">
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>Save</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="previousEntryDeleteButton">
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>Delete</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="previousEntryTextEdit"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTreeWidget" name="treeWidget">
        <property name="minimumSize">
         <size>
          <width>631</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>631</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="uniformRowHeights">
         <bool>true</bool>
        </property>
        <property name="sortingEnabled">
         <bool>false</bool>
        </property>
        <property name="animated">
         <bool>true</bool>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="expandsOnDoubleClick">
         <bool>false</bool>
        </property>
        <attribute name="headerCascadingSectionResizes">
         <bool>true</bool>
        </attribute>
        <attribute name="headerDefaultSectionSize">
         <number>400</number>
        </attribute>
        <attribute name="headerMinimumSectionSize">
         <number>300</number>
        </attribute>
        <column>
         <property name="text">
          <string>Attributes</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Values</string>
         </property>
        </column>
        <item>
         <property name="text">
          <string>Surname</string>
         </property>
         <property name="text">
          <string>Matikainen</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>First names</string>
         </property>
         <property name="text">
          <string>Jarkko Juhani</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Birthday</string>
         </property>
         <item>
          <property name="text">
           <string>year</string>
          </property>
          <property name="text">
           <string>1923</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>place</string>
          </property>
          <property name="text">
           <string>Suonenjoki</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>month</string>
          </property>
          <property name="text">
           <string>1</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>day</string>
          </property>
          <property name="text">
           <string>12</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Death</string>
         </property>
         <item>
          <property name="text">
           <string>year</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>month</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>fallen</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>day</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Profession</string>
         </property>
         <property name="text">
          <string>mv</string>
         </property>
         <property name="flags">
          <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Service</string>
         </property>
         <item>
          <property name="text">
           <string>Talvisota</string>
          </property>
          <property name="text">
           <string>True</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
          <item>
           <property name="text">
            <string>Regiments</string>
           </property>
           <property name="text">
            <string>ErP 33</string>
           </property>
          </item>
         </item>
         <item>
          <property name="text">
           <string>Rank</string>
          </property>
          <property name="text">
           <string>ylil 44</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Medals</string>
          </property>
          <property name="text">
           <string>Vm 1, Vm 2</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Jatkosota</string>
          </property>
          <property name="text">
           <string>True</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
          <item>
           <property name="text">
            <string>Regiments</string>
           </property>
           <property name="text">
            <string>JR 1, JR49</string>
           </property>
          </item>
         </item>
         <item>
          <property name="text">
           <string>Demobilization</string>
          </property>
          <item>
           <property name="text">
            <string>year</string>
           </property>
           <property name="text">
            <string>1944</string>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
          </item>
          <item>
           <property name="text">
            <string>place</string>
           </property>
           <property name="text">
            <string>Kotka</string>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
          </item>
          <item>
           <property name="text">
            <string>month</string>
           </property>
           <property name="text">
            <string>9</string>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
          </item>
          <item>
           <property name="text">
            <string>day</string>
           </property>
           <property name="text">
            <string>11</string>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
          </item>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Spouses</string>
         </property>
         <item>
          <property name="text">
           <string>Orvokki Pyörä</string>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
          </property>
          <item>
           <property name="text">
            <string>Wedding year</string>
           </property>
           <property name="text">
            <string>1947</string>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Children</string>
           </property>
           <item>
            <property name="text">
             <string>Woman's previous marriage</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Man's previous marriage</string>
            </property>
            <item>
             <property name="text">
              <string>Kaija</string>
             </property>
             <property name="text">
              <string>1946</string>
             </property>
             <property name="flags">
              <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
             </property>
            </item>
           </item>
           <item>
            <property name="text">
             <string>Current marriage</string>
            </property>
            <item>
             <property name="text">
              <string>Matti</string>
             </property>
             <property name="text">
              <string>1948</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Jussi</string>
             </property>
             <property name="text">
              <string>1950</string>
             </property>
            </item>
           </item>
          </item>
          <item>
           <property name="text">
            <string>Birthday</string>
           </property>
           <item>
            <property name="text">
             <string>year</string>
            </property>
            <property name="text">
             <string>1925</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>place</string>
            </property>
            <property name="text">
             <string>Iidesranta</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>month</string>
            </property>
            <property name="text">
             <string>7</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>day</string>
            </property>
            <property name="text">
             <string>4</string>
            </property>
           </item>
          </item>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Total childcount</string>
         </property>
         <property name="text">
          <string>3</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Children without mother data</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Hobbies</string>
         </property>
         <property name="text">
          <string>Kalastus, metsästys</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Address</string>
         </property>
         <property name="text">
          <string>Mäntykaari 3, Lietsala</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1524</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export</string>
     </property>
     <addaction name="actionCsv"/>
     <addaction name="actionJSON"/>
    </widget>
    <widget class="QMenu" name="menuImport">
     <property name="title">
      <string>Import</string>
     </property>
     <addaction name="actionFrom_txt_OCR"/>
    </widget>
    <addaction name="actionOpen_XML_for_analyze"/>
    <addaction name="actionSave_changes_to_xml"/>
    <addaction name="menuExport"/>
    <addaction name="menuImport"/>
   </widget>
   <widget class="QMenu" name="menuQuit">
    <property name="title">
     <string>Quit</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuQuit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionRun_analysis_for_all"/>
   <addaction name="actionCreate_a_new_Person"/>
  </widget>
  <action name="actionOpen_XML_for_analyze">
   <property name="text">
    <string>Open xml</string>
   </property>
  </action>
  <action name="actionCsv">
   <property name="text">
    <string>CSV</string>
   </property>
  </action>
  <action name="actionJSON">
   <property name="text">
    <string>JSON</string>
   </property>
  </action>
  <action name="actionOpen_txt">
   <property name="text">
    <string>Open txt</string>
   </property>
  </action>
  <action name="actionSave_changes_to_xml">
   <property name="text">
    <string>Save changes to xml</string>
   </property>
  </action>
  <action name="actionFrom_txt_OCR">
   <property name="text">
    <string>From txt (OCR)</string>
   </property>
  </action>
  <action name="actionRun_analysis_for_all">
   <property name="text">
    <string>Run analysis for all</string>
   </property>
   <property name="toolTip">
    <string>Run analysis for all entries in current file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
  <action name="actionRun_analysis_for_current">
   <property name="text">
    <string>Run analysis for current</string>
   </property>
   <property name="toolTip">
    <string>Run extraction for current person and update attributes</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionCreate_a_new_Person">
   <property name="text">
    <string>Create a  new Person</string>
   </property>
   <property name="toolTip">
    <string>Create a new person from rawtext</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EntriesListView</class>
   <extends>QListView</extends>
   <header>qtgui/entriesModels.h</header>
  </customwidget>
  <customwidget>
   <class>EntryTableView</class>
   <extends>QTableView</extends>
   <header>qtgui/entrytable.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
